lib/puppet/type/posix_acl.rb

```diff
@@ -232,19 +232,19 @@ def insync?(is)
            else
              raise ArgumentError, %(Unknown type "#{t}", expected "user", "group", "other" or "mask".)
            end
-      r << "#{a.shift}:" # Copy the "who".
+      r += "#{a.shift}:" # Copy the "who".
       p = a.shift
       if p =~ %r{[0-7]}
         p = p.oct
-        r << (p | 4 ? 'r' : '-')
-        r << (p | 2 ? 'w' : '-')
-        r << (p | 1 ? 'x' : '-')
+        r += (p | 4 ? 'r' : '-')
+        r += (p | 2 ? 'w' : '-')
+        r += (p | 1 ? 'x' : '-')
```

Comment:
I also wonder why this works. Shouldn't it be `&` instead of `|`? Because `0 | 4` is still `4` and `2 | 4` is `6`.

Reply:
yeah.... I will takle that with an acceptance test in another PR.